django_saml2_auth/views.py

```diff
@@ -117,11 +117,23 @@ def denied(r):
     return render(r, 'django_saml2_auth/denied.html')
 
 
-def _create_new_user(username, email, firstname, lastname):
-    user = user_model.objects.create_user(username, email)
-    user.first_name = firstname
-    user.last_name = lastname
-    groups = [Group.objects.get(name=x) for x in settings.SAML2_AUTH.get('NEW_USER_PROFILE', {}).get('USER_GROUPS', [])]
+def _create_new_user(user_identity):
+    attributes_map = settings.SAML2_AUTH.get('ATTRIBUTES_MAP', {})
+    user_name = user_identity[settings.SAML2_AUTH.get(
+        'ATTRIBUTES_MAP', {}).get('username', 'UserName')][0]
+    user = user_model.objects.create_user(user_name)
+    for key in attributes_map:
+        if key != 'username':
+            try:
+                if type(user_identity[attributes_map[key]]) is list:
+                    setattr(user, key, user_identity[attributes_map[key]][0])
+                else:
+                    setattr(user, key, user_identity[attributes_map[key]])
+            except KeyError:
+                pass
```

Comment:
the try-except here is more harmful than useful, since it will hide errors.

Reply:
I believe the attributes don't always come from haka. So keyerror is thrown and user can't log in if attribute is requested but missing.

Reply:
You're still having the try-except around a lot more code than just key lookup from `user_identity`. E.g. the setattr could also raise KeyError. Also, it's cleaner to use `var = ...get(something)` followed by `if var is not None`.